core/templates/pages/contributor-dashboard-page/question-opportunities/question-opportunities.component.spec.ts

```diff
@@ -0,0 +1,302 @@
+
+    spyOn($uibModal, 'open');
+    // The callFake is to avoid conflicts when testing modal calls.
+    spyOn(contributionOpportunitiesService, 'showRequiresLoginModal').and
+      .callFake(() => {});
+    ctrl.onClickSuggestQuestionButton('1');
+
+    expect($uibModal.open).not.toHaveBeenCalled();
+  });
+
+
+  it('should open select skill and skill difficulty modal when clicking' +
+    ' on suggesting question button', function() {
+    spyOn($uibModal, 'open').and.callThrough();
+    spyOn(userService, 'getUserInfoAsync').and.returnValue($q.resolve({
+      isLoggedIn: () => true
+    }));
+    ctrl.$onInit();
+    $rootScope.$apply();
+
+    ctrl.onClickSuggestQuestionButton('1');
+    $rootScope.$apply();
+
+    expect($uibModal.open).toHaveBeenCalled();
+  });
+
+  it('should open create question modal when creating a question', function() {
+    spyOn($uibModal, 'open').and.callThrough();
+    ctrl.createQuestion(
+      skillObjectFactory.createFromBackendDict({
+        id: '1',
+        description: 'test description',
+        misconceptions: [],
+        rubrics: [],
+        skill_contents: {
+          explanation: {
+            html: 'test explanation',
+            content_id: 'explanation',
+          },
+          worked_examples: [],
+          recorded_voiceovers: {
+            voiceovers_mapping: {}
+          }
+        },
+        language_code: 'en',
+        version: 3,
+      }), 1);
+    $rootScope.$apply();
+
+    expect($uibModal.open).toHaveBeenCalled();
+  });
+
+  it('should create a question when closing create question modal',
+    function() {
+      alertsService.clearWarnings();
+      var openSpy = spyOn($uibModal, 'open');
+      spyOn(userService, 'getUserInfoAsync').and.returnValue($q.resolve({
+        isLoggedIn: () => true
+      }));
+      ctrl.$onInit();
+      $rootScope.$apply();
+
+      openSpy.and.returnValue({
+        result: $q.resolve({
+          skill: skillObjectFactory.createFromBackendDict({
+            id: '1',
+            description: 'test description',
+            misconceptions: [],
+            rubrics: [],
+            skill_contents: {
+              explanation: {
+                html: 'test explanation',
+                content_id: 'explanation',
+              },
+              worked_examples: [],
+              recorded_voiceovers: {
+                voiceovers_mapping: {}
+              }
+            },
+            language_code: 'en',
+            version: 3,
+          }),
+          skillDifficulty: 1
+        })
+      });
+      ctrl.onClickSuggestQuestionButton('1');
+      openSpy.calls.reset();
+
+      spyOn(questionUndoRedoService, 'clearChanges');
+      openSpy.and.returnValue({
+        result: $q.resolve()
+      });
+      $rootScope.$apply();
+
+      expect(openSpy).toHaveBeenCalled();
+      expect(questionUndoRedoService.clearChanges).toHaveBeenCalled();
+    });
+
+  it('should create a question when dismissing create question modal',
```

Comment:
This sounds weird... Why is the question created when dismissing the modal?

Reply:
Done.